src/main/resources/templates/subjectSelection.html

```diff
@@ -0,0 +1,159 @@
+                                </div>
+                            </div>
+                        </div>
+                        <div class="modal-footer">
+                            <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+                            <button type="button" form="subjectForm" class="btn btn-primary" id="submitFormBtn">Submit</button>
+                        </div>
+                    </form>
+                </div>
+            </div>
+        </div>
+    </div>
+
+    <div class="modal fade" id="errorModal" tabindex="-1" role="dialog" aria-labelledby="errorModalLabel" aria-hidden="true">
+        <div class="modal-dialog" role="document">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h5 class="modal-title" id="errorModalLabel">Warning!</h5>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                </div>
+                <div class="modal-body" id="errorMessage">
+
+                </div>
+                <div class="modal-footer">
+                    <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
+
+<script src="https://code.jquery.com/jquery-3.6.0.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.7/umd/popper.min.js"></script>
+<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js"></script>
+
+
+<script>
+    $(document).ready(function() {
+        $('#addBtn').click(function() {
+            var selectedSubject = $('#subjects option:selected');
+            var selectedSubjectText = selectedSubject.text();
+            var selectedSubjectValue = selectedSubject.val();
+
+            var selectedYear = $('#schoolYear option:selected');
+            var selectedYearText = selectedYear.text();
+            var selectedYearValue = selectedYear.val();
+
+            var combinedValue = `${selectedSubjectValue}-${selectedYearValue}`;
+
+            $(`.selected-subject[data-combined-value="${combinedValue}"]`).remove();
+
+            if (selectedSubjectText && selectedYearText) {
+                var subjectHtml = `
+            <div class="selected-subject" data-combined-value="${combinedValue}">
+                <span>${selectedSubjectText} - ${selectedYearText}</span>
+                <button type="button" class="btn btn-link remove-btn"><i class="fas fa-trash"></i></button>
+                <input type="hidden" name="selectedValues" value="${combinedValue}">
+            </div>`;
+                $('#selectedSubjectsContainer').append(subjectHtml);
+            }
+        });
+
+        $('#selectedSubjectsContainer').on('click', '.remove-btn', function () {
```

Comment:
put all your js in js files like you already did. there can also be multiple functions in one js file, if they are for the same thing (i.e. everything that is in the modal)